crates/redis_interface/src/lib.rs

```diff
@@ -35,9 +35,57 @@ pub struct RedisConnectionPool {
     pub pool: fred::pool::RedisPool,
     config: RedisConfig,
     join_handles: Vec<fred::types::ConnectHandle>,
+    subscriber: RedisClient,
+    publisher: RedisClient,
     pub is_redis_available: Arc<atomic::AtomicBool>,
 }
 
+pub struct RedisClient {
+    inner: fred::prelude::RedisClient,
+}
+
+impl RedisClient {
+    pub async fn new(
+        config: fred::types::RedisConfig,
+        policy: fred::types::ReconnectPolicy,
+    ) -> CustomResult<Self, errors::RedisError> {
+        let client = fred::prelude::RedisClient::new(config);
+        client.connect(Some(policy));
+        client
+            .wait_for_connect()
+            .await
+            .into_report()
+            .change_context(errors::RedisError::RedisConnectionError)?;
+        Ok(Self { inner: client })
+    }
+    #[inline]
+    pub async fn subscribe(
+        &self,
+        channel: impl AsRef<str>,
+    ) -> CustomResult<usize, errors::RedisError> {
+        self.inner
+            .subscribe(channel.as_ref())
+            .await
+            .into_report()
+            .change_context(errors::RedisError::SubscribeError)
+    }
+    #[inline]
+    pub async fn publish(
+        &self,
+        channel: impl AsRef<str>,
+        key: impl AsRef<str>,
+    ) -> CustomResult<usize, errors::RedisError> {
+        self.inner
+            .publish(channel.as_ref(), key.as_ref())
+            .await
+            .into_report()
+            .change_context(errors::RedisError::SubscribeError)
+    }
+    #[inline]
+    pub fn on_message(&self) -> fred::prelude::AsyncStream<(String, fred::prelude::RedisValue)> {
```

Comment:
Can we prevent exposing `fred`'s types in return types? If it's not possible in this case, that's fine, not a necessity.